stormpath/express-stormpath
# Add support for preRegistrationHandler and preLoginHandler

Adds support for the `preRegistrationHandler` and `preLoginHandler`.

#### How to verify

1. Checkout this branch.
2. Link it (`$ npm link`).
3. Create a new test application using [this gist](https://gist.github.com/typerandom/94ed68b0ff541bf84597).
4. Link to our branch (`$ npm link express-stormpath`).
5. Start the application.
6. Navigate to `/register`.
7. Try and register with the email `test@test.com`.
8. Verify that you receive the error message `You're not allowed to register as 'test@test.com'.`.
9. Change the email to your own and set a really short password.
10. Verify that you receive an error stating that your password is too short.
11. Register for a new account.
12. Verify that the registration process works as expected.
13. Logout from your account.
14. Navigate to `/login`.
15. Try to login as `test@test.com`.
16. Verify that you receive the error message `You're not allowed to login as 'test@test.com'.`.
17. Enter an invalid username and password.
18. Verify that you receive a `username or password is invalid` error.
19. Enter the username and password that you previously registered with.
20. Verify that you are able to login as expected.

Fixes #299.
